### PR TITLE
905033-Added UG documentation of newly implemented Financial formulas in Calculate

### PR DESCRIPTION
**Pull request by ushasundarajan**
Added UG documentation of newly implemented financial formulas in Calculate that are listed below:

1.  NOMINAL
2.  MDURATION
3.  PDURATION
4.  COUPDAYS
5.  COUPDAYBS
6.  COUPDAYSNC
7.  COUPPCD
8.  COUPNCD
9.  COUPNUM

### CODE REVIEW

File: WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md [hunk 1]
```diff
@@ -744,3 +744,361 @@ _INTRATE(settlement, maturity, investment, redemption, [basis])_
 * Redemption    :The amount to be received at maturity.
 
 * Basis    :    The type of specifies the day count basis to used in the calculation.
+
+
+
+## NOMINAL
```
Comment (Manivannan-E) [inline WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md:750]: The NOMINAL function returns the nominal annual interest rate, given the effective annual interest rate and the number of compounding periods per year.

Syntax:

NOMINAL(effect_rate, npery)

Where:
effect_rate: The effective annual interest rate.
npery: The number of compounding periods per year.

Remarks:
npery is truncated to an integer.
If either argument is non-numeric, NOMINAL returns the #VALUE! error.
If effect_rate is less than or equal to 0 or if npery is less than 1, NOMINAL returns the #NUM! error.
The NOMINAL function is related to the EFFECT function, which calculates the effective annual interest rate based on the nominal rate and the number of compounding periods.

File: WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md [hunk 2]
```diff
@@ -744,3 +744,361 @@ _INTRATE(settlement, maturity, investment, redemption, [basis])_
...
+
+* If effect_rate ≤ 0 or if npery < 1, `NOMINAL` returns the `#NUM!` error value.
+
+* `NOMINAL` is related to `EFFECT` function. The relationship between `NOMINAL` and `EFFECT` is shown in the following equation:
+
+    `effective_rate = (1 + (nominal_rate / npery))^npery - 1`
+
+
+
+## MDURATION
```
Comment (Manivannan-E) [inline WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md:778]: @ushasundarajan 
The MDURATION function returns the modified Macaulay duration for a security with an assumed par value of $100.

Syntax:

MDURATION(settlement, maturity, coupon, yld, frequency, [basis])

Where:
settlement: The security's settlement date.
maturity: The security's maturity date.
coupon: The security's annual coupon rate.
yld: The security's annual yield.
frequency: The number of coupon payments per year (1 for annual, 2 for semiannual, 4 for quarterly).
basis (Optional): The day-count convention to use (default is 0: US (NASD) 30/360):
0 or omitted: US (NASD) 30/360
1: Actual/actual
2: Actual/360
3: Actual/365
4: European 30/360

Remarks:
Dates are stored as serial numbers (e.g., January 1, 1900, is represented as 1).
If settlement or maturity is not a valid date, MDURATION returns the #VALUE! error.
If yld or coupon is less than 0, or if frequency is not 1, 2, or 4, MDURATION returns the #NUM! error.
If basis is outside the range 0-4, MDURATION returns the #NUM! error.
If settlement is greater than or equal to maturity, MDURATION returns the #NUM! error.
MDURATION adjusts the Macaulay duration to account for changes in interest rates.

File: WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md [hunk 3]
```diff
@@ -744,3 +744,361 @@ _INTRATE(settlement, maturity, investment, redemption, [basis])_
...
+
+* If settlement is greater than or equal to maturity, it returns `#NUM!`.
+
+* Equation to calculate `MDURATION` is `Modified Duration = Macaulay Duration / (1 + (yld / frequency))`.
+
+
+
+## PDURATION
+
+The function `PDURATION` returns the number of periods required by an investment to reach a specified value.
```
Comment (Manivannan-E) [inline WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md:828]: @ushasundarajan 

The PDURATION function returns the number of periods required for an investment to reach a specified future value.

Syntax:

PDURATION(rate, pv, fv)

Where:
rate: The interest rate per period.
pv: The present value of the investment.
fv: The desired future value of the investment.

Remarks:
If rate is less than or equal to 0, or if pv or fv is less than or equal to 0, PDURATION returns the #NUM! error.
If any argument is non-numeric, PDURATION returns the #VALUE! error.
PDURATION calculates how many periods are needed for an investment to grow from its present value to the desired future value, based on the specified interest rate.

File: WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md [hunk 4]
```diff
@@ -744,3 +744,361 @@ _INTRATE(settlement, maturity, investment, redemption, [basis])_
...
+* If settlement is greater than or equal to maturity, it returns `#NUM!`.
+
+
+
+## COUPNUM
+
+The function `COUPNUM` returns the number of coupons payable between the settlement date and maturity date, rounded up to the nearest whole coupon.
+
+**Syntax:**
+
```
Comment (Manivannan-E) [inline WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md:1069]: @ushasundarajan 

The COUPNUM function returns the number of coupons payable between the settlement date and the maturity date, rounded up to the nearest whole coupon.

Syntax:

COUPNUM(settlement, maturity, frequency, [basis])

Where:

settlement: The security's settlement date.
maturity: The security's maturity date.
frequency: The number of coupon payments per year (1 for annual, 2 for semiannual, 4 for quarterly).
basis (Optional): The day count basis to use:
0 or omitted: US (NASD) 30/360
1: Actual/actual
2: Actual/360
3: Actual/365
4: European 30/360

Remarks:
Dates are stored as serial numbers (e.g., January 1, 1900, is represented as 1).
If settlement or maturity is not a valid date, COUPNUM returns #VALUE!.
If frequency is any number other than 1, 2, or 4, COUPNUM returns #NUM!.
If basis is outside the range 0-4, COUPNUM returns #NUM!.
If settlement is greater than or equal to maturity, COUPNUM returns #NUM!.
If any argument is non-numeric, COUPNUM returns #VALUE!.

File: WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md [hunk 5]
```diff
@@ -744,3 +744,361 @@ _INTRATE(settlement, maturity, investment, redemption, [basis])_
...
+* If frequency is any number other than 1, 2, or 4, `COUPPCD` returns `#NUM!`.
+
+* If basis is outside the range 0-4, it returns `#NUM!`.
+
+* If settlement is greater than or equal to maturity, it returns `#NUM!`.
+
+
+
+## COUPNCD
+
```
Comment (Manivannan-E) [inline WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md:1023]: @ushasundarajan 

The COUPNCD function returns the next coupon date after the settlement date.

Syntax:

COUPNCD(settlement, maturity, frequency, [basis])

Where:
settlement: The security's settlement date.
maturity: The security's maturity date.
frequency: The number of coupon payments per year (1 for annual, 2 for semiannual, 4 for quarterly).
basis (Optional): The day count basis to use:
0 or omitted: US (NASD) 30/360
1: Actual/actual
2: Actual/360
3: Actual/365
4: European 30/360
Remarks:

Dates are stored as serial numbers (e.g., January 1, 1900, is represented as 1).
If settlement or maturity is not a valid date, COUPNCD returns #VALUE!.
If frequency is any number other than 1, 2, or 4, COUPNCD returns #NUM!.
If basis is outside the range 0-4, COUPNCD returns #NUM!.
If settlement is greater than or equal to maturity, COUPNCD returns #NUM!.

File: WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md [hunk 6]
```diff
@@ -744,3 +744,361 @@ _INTRATE(settlement, maturity, investment, redemption, [basis])_
...
+* If frequency is any number other than 1, 2, or 4, `COUPDAYSNC` returns `#NUM!`.
+
+* If basis is outside the range 0-4, it returns `#NUM!`.
+
+* If settlement is greater than or equal to maturity, it returns `#NUM!`.
+
+
+
+## COUPPCD
```
Comment (Manivannan-E) [inline WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md:980]: @ushasundarajan 

The COUPPCD function returns the previous coupon date before the settlement date.

Syntax:

COUPPCD(settlement, maturity, frequency, [basis])

Where:
settlement: The security's settlement date.
maturity: The security's maturity date.
frequency: The number of coupon payments per year (1 for annual, 2 for semiannual, 4 for quarterly).
basis (Optional): The day count basis to use:
0 or omitted: US (NASD) 30/360
1: Actual/actual
2: Actual/360
3: Actual/365
4: European 30/360

Remarks:
Dates are stored as serial numbers (e.g., January 1, 1900, is represented as 1).
If settlement or maturity is not a valid date, COUPPCD returns #VALUE!.
If frequency is any number other than 1, 2, or 4, COUPPCD returns #NUM!.
If basis is outside the range 0-4, COUPPCD returns #NUM!.
If settlement is greater than or equal to maturity, COUPPCD returns #NUM!.

File: WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md [hunk 7]
```diff
@@ -744,3 +744,361 @@ _INTRATE(settlement, maturity, investment, redemption, [basis])_
...
+
+* If basis is outside the range 0-4, it returns `#NUM!`.
+
+* If settlement is greater than or equal to maturity, it returns `#NUM!`.
+
+
+
+## COUPDAYSNC
+
+The function `COUPDAYSNC` returns the number of days from the settlement date to the next coupon date.
```
Comment (Manivannan-E) [inline WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md:940]: @ushasundarajan 

The COUPDAYSNC function returns the number of days from the settlement date to the next coupon date.

Syntax:

COUPDAYSNC(settlement, maturity, frequency, [basis])

Where:

settlement: The security's settlement date.
maturity: The security's maturity date.
frequency: The number of coupon payments per year (1 for annual, 2 for semiannual, 4 for quarterly).
basis (Optional): The day count basis to use:
0 or omitted: US (NASD) 30/360
1: Actual/actual
2: Actual/360
3: Actual/365
4: European 30/360

Remarks:
Dates are stored as serial numbers (e.g., January 1, 1900, is represented as 1).
If settlement or maturity is not a valid date, COUPDAYSNC returns #VALUE!.
If frequency is any number other than 1, 2, or 4, COUPDAYSNC returns #NUM!.
If basis is outside the range 0-4, COUPDAYSNC returns #NUM!.
If settlement is greater than or equal to maturity, COUPDAYSNC returns #NUM!.

File: WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md [hunk 8]
```diff
@@ -744,3 +744,361 @@ _INTRATE(settlement, maturity, investment, redemption, [basis])_
...
+* If frequency is any number other than 1, 2, or 4, `COUPDAYS` returns `#NUM!`.
+
+* If basis is outside the range 0-4, `COUPDAYS` returns `#NUM!`.
+
+* If settlement is greater than or equal to maturity, `COUPDAYS` returns `#NUM!`.
+
+
+
+## COUPDAYBS
+
```
Comment (Manivannan-E) [inline WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md:897]: @ushasundarajan 
The COUPDAYBS function returns the number of days from the beginning of a coupon period until the settlement date.

Syntax:

COUPDAYBS(settlement, maturity, frequency, [basis])

Where:

settlement: The security's settlement date.
maturity: The security's maturity date.
frequency: The number of coupon payments per year (1 for annual, 2 for semiannual, 4 for quarterly).
basis (Optional): The day count basis to use:
0 or omitted: US (NASD) 30/360
1: Actual/actual
2: Actual/360
3: Actual/365
4: European 30/360

Remarks:
Dates are stored as serial numbers (e.g., January 1, 1900, is represented as 1).
If settlement or maturity is not a valid date, COUPDAYBS returns #VALUE!.
If frequency is any number other than 1, 2, or 4, COUPDAYBS returns #NUM!.
If basis is outside the range 0-4, COUPDAYBS returns #NUM!.
If settlement is greater than or equal to maturity, COUPDAYBS returns #NUM!.

File: WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md [hunk 9]
```diff
@@ -744,3 +744,361 @@ _INTRATE(settlement, maturity, investment, redemption, [basis])_
...
+* If any argument is non numeric, `PDURATION` returns the `#VALUE!` error value.
+
+* The `PDURATION` function calculates the number of periods required using the following equation:
+
+   `PDURATION = LOG(fv / pv) / LOG(1 + rate)`
+
+
+
+## COUPDAYS
+
```
Comment (Manivannan-E) [inline WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md:855]: @ushasundarajan 

The COUPDAYS function returns the number of days in the coupon period that contains the settlement date.

Syntax:

COUPDAYS(settlement, maturity, frequency, [basis])

Where:
settlement: The security's settlement date.
maturity: The security's maturity date.
frequency: The number of coupon payments per year (1 for annual, 2 for semiannual, 4 for quarterly).
basis (Optional): The day count basis to use:
0 or omitted: US (NASD) 30/360
1: Actual/actual
2: Actual/360
3: Actual/365
4: European 30/360

Remarks:
Dates are stored as serial numbers (e.g., January 1, 1900, is represented as 1).
If settlement or maturity is not a valid date, COUPDAYS returns #VALUE!.
If frequency is any number other than 1, 2, or 4, COUPDAYS returns #NUM!.
If basis is outside the range 0-4, COUPDAYS returns #NUM!.
If settlement is greater than or equal to maturity, COUPDAYS returns #NUM!.

File: WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md (hunk 1 above)
Comment (ushasundarajan) [reply WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md:750]: Changes updated.

File: WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md (hunk 2 above)
Comment (ushasundarajan) [reply WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md:776]: Changes updated

File: WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md (hunk 9 above)
Comment (ushasundarajan) [reply WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md:851]: Changes updated.

File: WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md (hunk 8 above)
Comment (ushasundarajan) [reply WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md:893]: Changes updated.

File: WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md (hunk 7 above)
Comment (ushasundarajan) [reply WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md:936]: @Manivannan-E , changes updated

File: WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md (hunk 6 above)
Comment (ushasundarajan) [reply WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md:976]: @Manivannan-E, changes updated.

File: WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md (hunk 5 above)
Comment (ushasundarajan) [reply WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md:1019]: @Manivannan-E, changes updated.

File: WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md (hunk 4 above)
Comment (ushasundarajan) [reply WindowsForms/Calculation-Engine/Supported-Formulas/Financial-Formulas.md:1065]: @Manivannan-E, changes updated.

**Review by Manivannan-E — APPROVED** [review]
Changes are fine.

**Pull request merged**